packages/xo-web/src/common/select-objects.js

```diff
@@ -166,40 +202,26 @@ class GenericSelect extends React.Component {
             options.push(getOption(object, container))
           })
         })
-      }
 
-      const objectsById = this._getObjectsById()
-      const addIfMissing = val => {
-        if (val != null && !(val in objectsById)) {
-          options.push({
-            disabled: true,
-            id: val,
-            label: val,
-            value: val,
-            xoItem: {
-              id: val,
-              removed: true,
-            },
+        // missing objects don't have a container
+        const missingObjects = objects['missingObjects']
+        if (missingObjects !== undefined) {
+          missingObjects.forEach(object => {
+            options.push(getOption(object))
           })
         }
       }
 
-      const values = this._getSelectedIds()
-      if (isArray(values)) {
-        forEach(values, addIfMissing)
-      } else {
-        addIfMissing(values)
-      }
-
+      options.map(option => {
+        if (option.xoItem.removed) {
+          option.disabled = true
```

Comment:
All the `removed` objects will be found in `missingObjects`, right? So you can do this in the `missingObjects.forEach` loop.

Reply:
Not all, just the missingObjects with containers.